specification/cognitiveservices/Language.AnalyzeText/models/common.models.tsp

```diff
@@ -0,0 +1,987 @@
+  length?: int32;
+}
+
+@doc("Parameters to query a knowledge base.")
+model AnswersOptions {
+  @doc("Exact QnA ID to fetch from the knowledge base, this field takes priority over question.")
+  qnaId?: int32;
+
+  @doc("User question to query against the knowledge base.")
+  question?: string;
+
+  @doc("Max number of answers to be returned for the question.")
+  top?: int32;
+
+  @doc("Unique identifier for the user.")
+  userId?: string;
+
+  @doc("Minimum threshold score for answers, value ranges from 0 to 1.")
+  @maxValue(1)
+  @minValue(0)
+  @clientName("confidenceThreshold")
+  confidenceScoreThreshold?: float64;
+
+  @doc("Context object with previous QnA's information.")
+  @clientName("answerContext")
+  context?: KnowledgeBaseAnswerContext;
+
+  @doc("Type of ranker to be used.")
+  @clientName("rankerKind")
+  rankerType?: RankerKind;
+
+  @doc("Filter QnAs based on given metadata list and knowledge base sources.")
+  filters?: QueryFilters;
+
+  @doc("To configure Answer span prediction feature.")
+  @clientName("shortAnswerOptions")
+  answerSpanRequest?: ShortAnswerOptions;
+
+  @doc("(Optional) Flag to enable Query over Unstructured Sources.")
+  includeUnstructuredSources?: boolean;
+}
+
+@doc("Type of ranker to be used")
+union RankerKind {
+  string,
+
+  @doc("Default ranker.")
+  Default: "Default ranker",
+
+  @doc("Question only ranker.")
+  Question: "Question only ranker",
+}
+
+@doc("Context object with previous QnA's information.")
+model KnowledgeBaseAnswerContext {
+  @doc("Previous turn top answer result QnA ID.")
+  previousQnaId: int32;
+
+  @doc("Previous user query.")
+  @clientName("previousQuestion")
+  previousUserQuery?: string;
+}
+
+@doc("filters over knowledge base.")
+model QueryFilters {
+  @doc("filters over knowledge base.")
+  metadataFilter?: MetadataFilter;
+
+  @doc("filters over knowledge base.")
+  sourceFilter?: SourceFilter;
+
+  @doc("Logical operation used to join metadata filter with source filter.")
+  logicalOperation?: LogicalOperationKind = LogicalOperationKind.AND;
+}
+
+@doc("Find QnAs that are associated with the given list of metadata.")
+model MetadataFilter {
+  @doc("List of metadata.")
+  metadata?: Array<MetadataRecord>;
+
+  @doc("Operation used to join metadata filters.")
+  logicalOperation?: LogicalOperationKind = LogicalOperationKind.AND;
+}
+
+@doc("Object to provide the key value pair for each metadata.")
+model MetadataRecord {
+  @doc("Metadata Key from Metadata dictionary used in the QnA.")
+  key: string;
+
+  @doc("Metadata Value from Metadata dictionary used in the QnA.")
+  value: string;
+}
+
+@doc("Find QnAs that are associated with any of the given list of sources in knowledge base.")
+model SourceFilter is Array<string>;
+
+#suppress "@azure-tools/typespec-azure-core/no-closed-literal-union" // defined as closed enum
+@doc("Set to 'OR' or 'AND' for using corresponding logical operation.")
+union LogicalOperationKind {
```

Comment:
You can use `enum` for Closed enum.

Reply:
for `enum` we would have to suppress 2 warnings
```
#suppress "@azure-tools/typespec-azure-core/no-enum"
#suppress "@azure-tools/typespec-azure-core/use-extensible-enum"
```
 with `union` we only have to suppress one
`#suppress "@azure-tools/typespec-azure-core/no-closed-literal-union"`